R/lancelot.R

```diff
@@ -2774,29 +2774,18 @@ lancelot_n_groups <- function() {
 ##' @export
 lancelot_check_severity <- function(pars) {
   check_parameters <- function(p_step, rel_p) {
-    vapply(
-      seq_len(pars$n_groups),
-      function(i) {
-        if (is.null(pars[[rel_p]])) {
-          stop(sprintf("Parameter '%s' is missing", rel_p))
-        } else if (is.null(pars[[p_step]])) {
-          stop(sprintf("Parameter '%s' is missing", p_step))
-        }
+    if (is.null(pars[[rel_p]])) {
+      stop(sprintf("Parameter '%s' is missing", rel_p))
+    } else if (is.null(pars[[p_step]])) {
+      stop(sprintf("Parameter '%s' is missing", p_step))
+    }
```

Comment:
These tests were being applied over $i$ even though they do not depend upon $i$. Thus they were being applied $n_groups$ times. Here we are now applying the same tests just once